src/components/application_manager/src/help_prompt_manager_impl.cc

```diff
@@ -66,38 +70,41 @@ bool HelpPromptManagerImpl::AddCommand(
     return false;
   }
 
+  const smart_objects::SmartObject& commands = command[strings::vr_commands];
+  if (commands.empty()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands array is empty");
+    return false;
+  }
+
   sync_primitives::AutoLock lock(vr_commands_lock_);
   auto it = std::find_if(
       vr_commands_.begin(),
       vr_commands_.end(),
       [cmd_id](const VRCommandPair& pair) { return pair.first == cmd_id; });
 
   if (vr_commands_.end() != it) {
-    LOG4CXX_DEBUG(logger_, "Commands with id:" << cmd_id << " already exists");
+    LOG4CXX_DEBUG(logger_, "Command with id:" << cmd_id << " already exists");
     return false;
   }
 
-  const smart_objects::SmartObject& commands = command[strings::vr_commands];
   const size_t count_new_commands = commands.length();
   const bool limit_exceeded =
       kLimitCommand <= GetCommandsCount(vr_commands_.end());
 
-  LOG4CXX_DEBUG(logger_, "Will be added " << count_new_commands << " commands");
+  LOG4CXX_DEBUG(logger_,
+                "Will be added first of " << count_new_commands << " commands");
 
   smart_objects::SmartObjectSPtr vr_item =
       std::make_shared<smart_objects::SmartObject>(
-          smart_objects::SmartType_Array);
-  smart_objects::SmartArray& ar_vr_cmd = *(vr_item->asArray());
+          smart_objects::SmartType_String);
   smart_objects::SmartArray& ar_cmd = *(commands.asArray());
-  ar_vr_cmd.reserve(count_new_commands);
-  ar_vr_cmd.insert(
-      ar_vr_cmd.end(), ar_cmd.begin(), ar_cmd.begin() + count_new_commands);
+  *vr_item = ar_cmd.front();
   vr_commands_.push_back(std::make_pair(cmd_id, vr_item));
```

Comment:
Seems this whole section could be reduced to two lines:
```
  smart_objects::SmartObjectSPtr vr_item =
      std::make_shared<smart_objects::SmartObject>((commands.asArray())->front());
  vr_commands_.push_back(std::make_pair(cmd_id, vr_item));
```

Reply:
@jacobkeeler good point. Updated in https://github.com/smartdevicelink/sdl_core/pull/2557/commits/445e711459cce157e63e355f3e20df300aa7a963